src/components/DataTypes/array.tsx

```diff
@@ -49,68 +45,44 @@ interface StringFilter extends JSONSchema {
 	}>;
 }
 
-export const decodeFilter = (
-	filter: StringFilter,
-): {
-	field: string;
-	operator: OperatorSlug;
-	value: string;
-} | null => {
-	const operator = filter.title;
-	const properties = findInObject(filter, 'properties');
-	const [firstPropKey] = Object.keys(properties);
-	const field = firstPropKey;
-	const value =
-		findInObject(filter, 'const') ?? findInObject(filter, 'pattern');
-
-	return {
-		field,
-		operator,
-		value,
-	};
-};
-
 export const createFilter = (
 	field: string,
-	operator: OperatorSlug,
+	operator: Operator,
 	value: any,
 	schema: JSONSchema,
+	recursive: boolean = false,
+	refScheme?: string,
+	isFullTextSearch?: boolean,
 ): JSONSchema => {
 	const { title } = schema;
-	const base: StringFilter = {
-		$id: randomString(),
-		title: operator,
-		description: getJsonDescription(
-			title || field,
-			operators[operator].getLabel(schema),
-			value,
-		),
-		type: 'object',
-	};
-
-	const filter = getFilter(schema, value);
-
-	if (operator === 'contains') {
-		return Object.assign(base, {
-			properties: {
-				[field]: {
-					type: 'array',
-					...filter,
-				},
-			},
-			required: [field],
-		});
-	}
+	const operatorSlug = operator.slug;
+	const base = getBase(
+		recursive,
+		operator,
+		title,
+		field,
+		value,
+		refScheme,
+	) as ArrayFilter;
 
-	if (operator === 'not_contains') {
+	const filter = getFilter(field, schema, value, operator, isFullTextSearch);
+	if (
+		operatorSlug === 'not_contains' &&
+		!!schema.items &&
+		typeof schema.items === 'object' &&
+		'type' in schema.items &&
+		!('properties' in schema.items)
+	) {
 		return Object.assign(base, {
 			anyOf: [
 				{
 					properties: {
 						[field]: {
 							type: 'array',
-							not: {
-								...filter,
+							contains: {
+								not: {
```

Comment:
```suggestion
							not: {
								contains: {
```
See: https://www.jsonschemavalidator.net/s/47xdkAWd

Reply:
with ajv 6.12 contains.not seems to be the right one https://codesandbox.io/s/ajv-playground-forked-c4lk06?file=/src/index.js